### PR TITLE
Showing real transaction hash while sending funds

### PR DESCRIPTION
**Pull request by xitronix**

### CODE REVIEW

File: universal-login-wallet/src/ui/react/Modals/Transfer/ModalTransfer.tsx [hunk 1]
```diff
@@ -16,9 +16,11 @@ const ModalTransfer = () => {
 
   const transferService = new TransferService(sdk, applicationWallet);
   const onGenerateClick = async () => {
-    modalService.showModal('waitingForTransfer');
     try {
-      await transferService.transfer(transferDetalis);
+      const {waitToBeSuccess, waitForTransactionHash} = await transferService.transfer(transferDetalis);
+      const {transactionHash} = await waitForTransactionHash();
+      modalService.showModal('waitingForTransfer', transactionHash);
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/ui/react/Modals/Transfer/ModalTransfer.tsx]: We should show dialog before we have a hash (but without a hash), and update once we have a hash

File: universal-login-wallet/test/integration/pages/DashboardPage.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@ export default class DashboardPage {
   }
 
   async waitForHideModal() {
-    await waitForUI(this.wrapper, () => !this.wrapper.exists('.modal-wrapper'));
```
Comment (marekkirejczyk) [inline universal-login-wallet/test/integration/pages/DashboardPage.ts]: why 4000?

File: universal-login-wallet/src/ui/react/Modals/ModalWaitingFor.tsx [hunk 3]
```diff
@@ -20,7 +25,7 @@ const ModalWaitingFor = ({action}: {action: string}) => {
         <div className="modal-pending-section">
           <ProgressBar className="modal-pending-loader"/>
           <h3 className="modal-section-title transaction-hash-title">Transaction status</h3>
-          <p className="txn-hash-text">0x5a63…f3a68a</p>
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/ui/react/Modals/ModalWaitingFor.tsx]: - Do we need to add ... like `0x1234...FF89`?
- Can we make it a link to etherscan?

File: universal-login-wallet/src/ui/react/Modals/Modal.tsx [hunk 4]
```diff
@@ -51,13 +51,13 @@ const Modal = () => {
     case 'waitingForDeploy':
       return (
         <ModalWrapper modalClassName="jarvis-modal">
-          <ModalWaitingFor action={'Txn pending'} />
+          <ModalWaitingFor action={'Txn pending'} transactionHash={'0x5a63…f3a68a'}/>
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/ui/react/Modals/Modal.tsx]: Magic value: `0x5a63…f3a68a`?

Comment (xitronix) [reply universal-login-wallet/src/ui/react/Modals/Modal.tsx]: It was hard coded into modal and then extracted into ModalParameters. It is placeholder for future real transactionHash from deployment

File: universal-login-wallet/src/ui/react/Modals/ModalWaitingFor.tsx (hunk 3 above)
Comment (xitronix) [reply universal-login-wallet/src/ui/react/Modals/ModalWaitingFor.tsx]: Yes, we can. Something like this? 
```<a href=`https://etherscan.io/tx/${transactionHash}`" target="_blank">{formatTransactionHash(transactionhash)}</a> ```

File: universal-login-wallet/test/integration/pages/DashboardPage.ts (hunk 2 above)
Comment (xitronix) [reply universal-login-wallet/test/integration/pages/DashboardPage.ts]: I had timeout error on my laptop (default is 3000 and test had time ~3300)

File: universal-login-wallet/src/ui/react/Modals/Modal.tsx (hunk 4 above)
Comment (marekkirejczyk) [reply universal-login-wallet/src/ui/react/Modals/Modal.tsx]: ok

File: universal-login-wallet/src/ui/react/Modals/ModalWaitingFor.tsx [hunk 5]
```diff
@@ -20,7 +26,7 @@ const ModalWaitingFor = ({action}: {action: string}) => {
...
+          <p className="txn-hash-text"><a href={`https://${chainName.toLowerCase().trim()}.etherscan.io/tx/${transactionHash}`} target="_blank">{transactionHash}</a> </p>
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/ui/react/Modals/ModalWaitingFor.tsx]: extract `linkToExplorer(chainName: string, hash:string)`

note: hash can be account or transaction

**Pull request merged**